plugins/available/ruby.plugin.bash

```diff
@@ -1,16 +1,17 @@
+# shellcheck shell=bash
 cite about-plugin
 about-plugin 'ruby and rubygems specific functions and settings'
 
 # Make commands installed with 'gem install --user-install' available
 # ~/.gem/ruby/${RUBY_VERSION}/bin/
-if which ruby >/dev/null && which gem >/dev/null; then
-  pathmunge "$(ruby -e 'print Gem.user_dir')/bin" after
+if _command_exists ruby && _command_exists gem; then
+	pathmunge "$(ruby -e 'print Gem.user_dir')/bin" after
 fi
 
-function remove_gem {
-  about 'removes installed gem'
-  param '1: installed gem name'
-  group 'ruby'
+function remove_gem() {
+	about 'removes installed gem'
+	param '1: installed gem name'
+	group 'ruby'
 
-  gem list | grep $1 | awk '{ print $1; }' | xargs sudo gem uninstall
+	gem list | grep "${1?}" | awk '{ print $1; }' | xargs sudo gem uninstall
```

Comment:
re `"${1?}"`

* Maybe check the value _before_ invoking `gem list`
```
: "${1?}"
gem list | grep "${1}" | awk '{ print $1; }' | xargs sudo gem uninstall
```

* Maybe also check for empty values vs just unset:
```
: "${1:?}"
```

* Maybe add a more useful error message:
```
: "${1:?no gem name provided}"
```

Even with this, the generated error message would be:
```
bash: 1: no gem name provided
```
Which feels a _tad_ cryptic but does get the message across.

Reply:
The `${1?}` usage validates the parameter *before* invoking any commands, so gem list won't be executed if it fails.

There's currently no handling of this parameter being empty at all, but what I often do is something like `"${1?${FUNCNAME[0]}: no gem name provided}"` which would result in:
```
bash: 1: remove_gem: no gem name provided
```

Reply:
How's this look now?